### PR TITLE
[PR] Generate cypher and length disasm

### PR DESCRIPTION
**Pull request by spolowy**

### CODE REVIEW

File: srcs/virus/generate_cypher.c [hunk 1]
```diff
@@ -256,3 +230,45 @@ bool		generate_decypher(char *buffer, uint64_t seed, size_t size)
 	generate_unshuffler(frame.ptr, seed, frame.size);
 	return true;
 }
+
```
Comment (nicolhaq) [inline srcs/virus/generate_cypher.c]: normale se code entre comentaire?

File: includes/polymorphism.h [hunk 2]
```diff
@@ -19,8 +19,8 @@
 
 uint64_t	polymorphic_seed_engine(uint64_t son_seed[2], uint64_t client_id);
 
-bool		generate_cypher(char *buffer, uint64_t seed, size_t size);
-bool		generate_decypher(char *buffer, uint64_t seed, size_t size);
+bool		generate_cypher(void *buffer, uint64_t seed, size_t size);
```
Comment (grumbach) [inline includes/polymorphism.h]: OUI! OUI! OUI! Depuis le temps que ça m'irrite!

File: srcs/virus/generate_cypher.c [hunk 3]
```diff
@@ -215,27 +189,27 @@ static struct safe_ptr    generate_loop_frame(char *buffer, size_t size)
 	if (size < sizeof(footer) + sizeof(header))
 		return (struct safe_ptr){NULL, 0};
 
-	char	*remaining_buffer = buffer + sizeof(header);
+	void	*remaining_buffer = buffer + sizeof(header);
 	size_t	remaining_size    = size - sizeof(footer) - sizeof(header);
 
-	int16_t *rel_cypher_end = (int16_t *)&header[13];
-	int16_t *rel_cypher     = (int16_t *)&footer[12];
+	uint64_t	*rel_cypher_end = (uint64_t*)&header[13];
```
Comment (grumbach) [inline srcs/virus/generate_cypher.c]: Wait! 
writing uint64_t to 32 bit location!!!
```
                    v~~~~~~~~~~~~~~~
0x0f, 0x84, 0x13, 0x00, 0x00, 0x00,      /*     jz cypher_end      */
		0x48, 0x8b, 0x07                    /*     mov rax, [rdi]     */
```

Comment (grumbach) [reply srcs/virus/generate_cypher.c]: this line!
https://github.com/spolowy/War/blob/df9451ab5d175f775e77e2837a10abb6743f931c/srcs/virus/generate_cypher.c#L203

Comment (grumbach) [reply srcs/virus/generate_cypher.c]: make it 32bit and check for overflows on 32bits! 
https://github.com/spolowy/War/blob/df9451ab5d175f775e77e2837a10abb6743f931c/srcs/virus/generate_cypher.c#L199

**Pull request merged**